src/raftstore/store/worker/pd.rs

```diff
@@ -174,6 +182,57 @@ impl<T: PdClient> Runner<T> {
             error!("report split failed {:?}", e);
         }
     }
+
+    fn handle_validate_peer(&self, local_region: metapb::Region, peer: metapb::Peer) {
+        metric_incr!("pd.get_region");
+        match self.pd_client.get_region(local_region.get_start_key()) {
+            Ok(mut pd_region) => {
+                metric_incr!("pd.get_region");
+                if pd_region == local_region {
+                    // Local region info is as fresh as pd, which means
+                    // local peer is still in the region, let it be there.
+                    info!("peer {:?} is still valid because local region info is as fresh as pd",
+                          peer);
+                    return;
+                }
+                if !(pd_region.get_start_key() == local_region.get_start_key()) {
+                    // The region [start_key, ...) is missing in pd currently. It's probably
+                    // that a pending region split is happenning right now and that region
+                    // doesn't report it's heartbeat(with updated region info) yet.
+                    // We should sit tight and try another get_region task later.
+                    warn!("peer {} fails to get region info from pd with start key: {:?}, retry \
+                           later",
+                          peer.get_id(),
+                          local_region.get_start_key());
+                    return;
+                }
+                let valid = pd_region.get_peers().into_iter().any(|p| p.to_owned() == peer);
+                if !valid {
+                    // Peer is not a member of this region anymore. Probably it's removed out.
+                    // Send it a raft massage to destroy it since it's obsolete.
+                    info!("peer {} is not a valid member of region {:?}. To be destroyed soon.",
+                          peer.get_id(),
+                          pd_region);
+                    let mut message = RaftMessage::new();
```

Comment:
More info on where is the init function? After a global search in the whole repo, I couldn't find it.